jdaviz/configs/imviz/plugins/coords_info/coords_info.py

```diff
@@ -493,9 +487,18 @@ def _image_viewer_update(self, viewer, x, y):
                     image, arr, x, y, viewer
                 )
 
-                # We don't want to convert for things like moment maps
-                if str(u.Unit(unit).physical_type) not in ("spectral flux density",
-                                                           "surface brightness"):
+                # We don't want to convert for things like moment maps, so check physical type
+                # If unit is flux per pix2, the type will be 'unknown' rather
+                # than surface brightness, so have to multiply the pix2 part out
+                # and check if the numerator is a spectral flux density
+                if check_if_unit_is_per_solid_angle(unit, return_unit=True) == u.pix*u.pix:
```

Comment:
Should we move `PIX2` higher up and re-use it throughout the package instead of `u.pix*u.pix`?

Reply:
I'll do this after all these PRs are merged to avoid conflicts